coalib/testing/BaseTestHelper.py

```diff
@@ -1,45 +1,43 @@
-from coala_utils.Comparable import Comparable
+import unittest
 
+from coala_utils.Comparable import Comparable
 
-def _assert_comparable_equal(observed_result, expected_result):
-    """
-    Compares individual fields of the result objects using
-    `__compare_fields__` of `coala_utils.Comparable` class
-    and yields messages in case of an attribute mismatch.
-    """
 
-    if not len(observed_result) == len(expected_result):
-        assert observed_result == expected_result, '%s != %s' % (
-            observed_result, expected_result)
-
-    messages = ''
-    for observed, expected in zip(observed_result, expected_result):
-        if (isinstance(observed, Comparable) and
-            isinstance(expected, Comparable)) and (type(observed) is
-                                                   type(expected)):
-            for attribute in type(observed).__compare_fields__:
-                try:
-                    assert getattr(observed, attribute) == getattr(
-                        expected, attribute), (
-                        '{} mismatch: {}, {} != {}, {}'.format(
-                            attribute,
-                            observed.origin, observed.message,
-                            expected.origin, expected.message))
-                except AssertionError as ex:
-                    messages += (str(ex) + '\n\n')
-        else:
-            assert observed_result == expected_result, '%s != %s' % (
-                observed_result, expected_result)
-
-    if messages:
-        raise AssertionError(messages)
-
-
-class BaseTestHelper:
+class BaseTestHelper(unittest.TestCase):
     """
     This is a base class for all Bears' tests of coala's testing API.
     """
 
+    def _assert_comparable_equal(self,
+                                 observed_result,
+                                 expected_result):
+        """
+        Compares individual fields of the result objects using
+        `__compare_fields__` of `coala_utils.Comparable` class
+        and raises messages in case of an attribute mismatch.
+        """
+
+        if not len(observed_result) == len(expected_result):
+            self.assertEqual(observed_result, expected_result)
+
+        messages = []
+        for observed, expected in zip(observed_result, expected_result):
+            if (isinstance(observed, Comparable) and
+                isinstance(expected, Comparable)) and (
+                    Comparable.__eq__(expected, observed)):
+                for attribute in observed.__compare_fields__:
+                    try:
+                        self.assertEqual(getattr(observed, attribute),
+                                         getattr(expected, attribute),
+                                         msg='{} mismatch.'.format(attribute))
+                    except AssertionError as ex:
+                        messages.append(str(ex))
```

Comment:
this makes the test suite reliant on pytest output, which differs whenever pytest finds a way to improve their messages.